kernel/generic/src/ipc/sysipc.c

```diff
@@ -900,18 +900,24 @@ sysarg_t sys_ipc_irq_unsubscribe(sysarg_t cap)
 
 /** Syscall connect to a task by ID (32 bits)
  *
- * @return Phone id on success, or negative error code.
+ * @return Error code.
  *
  */
-sysarg_t sys_ipc_connect_kbox(sysarg64_t *uspace_taskid)
+sysarg_t sys_ipc_connect_kbox(sysarg64_t *uspace_taskid, cap_handle_t *uspace_phone)
 {
 #ifdef CONFIG_UDEBUG
 	sysarg64_t taskid;
+	cap_handle_t phone;
 	int rc = copy_from_uspace(&taskid, uspace_taskid, sizeof(sysarg64_t));
-	if (rc != 0)
-		return (sysarg_t) rc;
+	if (rc == EOK) {
+		rc = ipc_connect_kbox((task_id_t) taskid, &phone);
+	}
+	if (rc == EOK) {
+		rc = copy_to_uspace(uspace_phone, &phone, sizeof(cap_handle_t));
+		// FIXME: Clean up phone on failure.
```

Comment:
sys_ipc_hangup() comes to mind here. Even though I'm planning changes in this area that would destroy the phone capability non-lazily.